stl/inc/ranges

```diff
@@ -77,6 +77,21 @@ namespace ranges {
     template <class _Ty, size_t _Extent>
         requires (_Extent != dynamic_extent)
     inline constexpr auto _Compile_time_max_size<const span<_Ty, _Extent>> = _Extent;
+
+#ifdef __cpp_lib_byte
+    using _Elements_alloc_type = byte;
+#else
+    using _Elements_alloc_type = char;
+#endif

```

Comment:
Hmm, perhaps we should be consistent on whether there's workaround when `__cpp_lib_byte` is not defined, see https://github.com/microsoft/STL/pull/4337#discussion_r1462569445.

Reply:
Yeah, this is different than our previous `__cpp_lib_byte` checks, and is ODR-violation-ish.

Really we need to push for the WinSDK to be fixed, but in the meantime we should definitely think about what to do here.